src/app/modules/react-migration/react-migration.controller.ts

```diff
@@ -38,25 +42,61 @@ export const ADMIN_COOKIE_OPTIONS_WITH_EXPIRY = {
 
 const logger = createLoggerWithLabel(module)
 
-const serveFormReact: ControllerHandler = (_req, res) => {
-  const reactFrontendPath = path.resolve('dist/frontend')
-  logger.info({
-    message: 'serveFormReact',
-    meta: {
-      action: 'routeReact.serveFormReact',
-      __dirname,
-      cwd: process.cwd(),
-      reactFrontendPath,
-    },
-  })
-  return (
-    res
-      // Prevent index.html from being cached by browsers.
-      .setHeader('Cache-Control', 'no-cache')
-      .sendFile(path.join(reactFrontendPath, 'index.html'))
-  )
+const reactFrontendPath = path.resolve('dist/frontend')
+const reactHtml = readFileSync(path.join(reactFrontendPath, 'index.html'), {
+  encoding: 'utf8',
+})
+
+type MetaTags = {
+  title: string
+  description: string
+  image: string
+}
+const replaceWithMetaTags = ({
+  title,
+  description,
+  image,
+}: MetaTags): string => {
+  return reactHtml
+    .replace(/(__OG_TITLE__)/g, DOMPurify.sanitize(title))
+    .replace(/(__OG_DESCRIPTION__)/g, DOMPurify.sanitize(description))
+    .replace(/(__OG_IMAGE__)/g, DOMPurify.sanitize(image))
```

Comment:
Few extra minor nits and questions (sorry! 😅 )
* brackets are not needed in the regex patterns
* Does `DOMPurify.sanitize()` have different mechanisms to encode raw markup on one hand, and attributes on the other hand? some of the markup injection we are doing here is markup (`<title>` tag) while the rest are attributes.
* Sanitization might be over kill for this (not sure 🤔), and [html-escaping](https://www.npmjs.com/package/html-escaper) might have been enough

BTW I realized my previous comment was wrong: considering the form title is user-supplied, we don't know what content it will have exactly, so escaping or sanitising was indeed the right thing to do 👍

Reply:
Just adding a comment from our earlier discussion. In this use case, doing escaping is better than sanitizing, because the values are not meant to contain HTML at all.

So last TODO for this PR is to replace `DOMPurify.sanitize()` with `html-escaper.escape()`